code/modules/recycling/wrapped_package.dm

```diff
@@ -0,0 +1,283 @@
+		
+	update_icon()
+
+/obj/item/parcel/on_update_icon()
+	. = ..()
+	if(w_class < ITEM_SIZE_NO_CONTAINER)
+		icon_state = "[icon_state_prefix]_[clamp(round(w_class), ITEM_SIZE_MIN, ITEM_SIZE_HUGE)]"
+	else
+		//If its none of the smaller items, default to crate-sized package
+		icon_state = "[icon_state_prefix]_crate"
+		//Try to see if we got a better icon state for whatever we contain
+		if(length(contents))
+			if(istype(contents?[1], /obj/structure/closet) && !istype(contents?[1], /obj/structure/closet/crate))
+				icon_state = "[icon_state_prefix]_closet"
+			else if(ishuman(contents?[1]))
+				icon_state = "[icon_state_prefix]_human"
+
+	//Apply the sorting tag icon on top of our sprite. The extension does it for us.
+	var/datum/extension/sorting_tag/S = get_extension(src, /datum/extension/sorting_tag)
+	if(S)
+		S.apply_tag_overlay()
+	
+	//Put a label on if we added a note
+	if(length(attached_note) && icon_state != "[icon_state_prefix]_1")
+		var/off_x = 0
+		var/off_y = 0
+		if(icon_state == "[icon_state_prefix]_5")
+			off_y = -1
+		else if(icon_state == "[icon_state_prefix]_crate")
+			off_x = rand(-8, 6)
+			off_y = -3
+		else if(icon_state == "[icon_state_prefix]_closet")
+			off_x = 2
+			off_y = rand(-6, 11)
+		else if(icon_state == "[icon_state_prefix]_human")
+			off_x = rand(-3, 3)
+			off_y = rand(-6, 11)
+
+		var/image/I = image('icons/obj/items/storage/deliverypackage.dmi', "delivery_label", pixel_x = off_x, pixel_y = off_y)
+		add_overlay(I)
+	
+/obj/item/parcel/examine(mob/user, distance)
+	. = ..()
+	if(distance < 3)
+		var/datum/extension/sorting_tag/S = get_extension(src, /datum/extension/sorting_tag)
+		if(S)
+			to_chat(user, S.tag_description())
+		if(length(attached_note))
+			to_chat(user, "It has a note attached, which reads:'[attached_note]'.")
+
+/obj/item/parcel/get_mechanics_info()
+	. = ..()
+	. += "<BR/>It can be opened by applying any sharp item, with help intent."
+	. += "<BR/>It can opened by using it while held, if its small enough."
+
+/obj/item/parcel/proc/make_parcel(var/atom/movable/AM, var/mob/user)
+	if(!is_type_in_list(AM, get_whitelist()))
+		log_warning("[src] ([x], [y], [z]) was passed an invalid atom type to contain[istype(AM, /datum)? " '[AM.type]'" : ", value: [AM]"].")
+		return
+	var/parcel_name
+	var/parcel_size
+	var/obj/O = AM
+
+	if(istype(O))
+		parcel_size = round(O.w_class)
+	else if(ismob(AM))
+		var/mob/M = AM
+		parcel_size = round(M.mob_size)
+	else
+		CRASH("Make parcel got passed an invalid atom type '[AM?.type]'.")
+
+	//Name the parcel based on its size
+	switch(parcel_size)
+		if(ITEM_SIZE_TINY)
+			parcel_name = "tiny [initial(name)]"
+		if(ITEM_SIZE_SMALL)
+			parcel_name = "small [initial(name)]"
+		if(ITEM_SIZE_NORMAL)
+			parcel_name = "normal-sized [initial(name)]"
+		if(ITEM_SIZE_LARGE)
+			parcel_name = "large [initial(name)]"
+		if(ITEM_SIZE_HUGE, ITEM_SIZE_GARGANTUAN)
+			parcel_name = "huge [initial(name)]"
+		else
+			parcel_name = "enormous [initial(name)]"
```

Comment:
I wouldn't call that the same units.

```dm
#define ITEM_SIZE_TINY           1
#define ITEM_SIZE_SMALL          2
#define ITEM_SIZE_NORMAL         3
#define ITEM_SIZE_LARGE          4
#define ITEM_SIZE_HUGE           5
#define ITEM_SIZE_GARGANTUAN     6
#define ITEM_SIZE_NO_CONTAINER   10 // Use this to forbid item from being placed in a container.
#define ITEM_SIZE_STRUCTURE      20
```
```dm
#define MOB_SIZE_LARGE  		40
#define MOB_SIZE_MEDIUM 		20
#define MOB_SIZE_SMALL 		10
#define MOB_SIZE_TINY 		5
#define MOB_SIZE_MINISCULE	1
```
Honestly, I'd have made small mobs the size of a normal/large sized object. 

But the problem is that the w_class is also used to prevent things from being inserted into storage as well as give its size. So when you don't want something to be inserted into storage, you also give up on any meaningful size value. Which is kinda annoying when you're trying to give a visual representation of a wrapped object. 

So I just sorta gave up because there's no good solution to this currently short of making some type checking mess.

Reply:
> > I wouldn't call that the same units.
> 
> We've had this conversation before. If you disagree with the scale of sizes, sure, but as they are currently, they are intended to be the same units. If you look at the places in code where both values are used, they are equivalent. You could make `w_class` and `mob_size` into a single var on `/atom/movable` like `atom_size` and it wouldn't impact much in the code.
> 
> > So when you don't want something to be inserted into storage, you also give up on any meaningful size value.
> 
> Make it an atom flag.

Yeah, I remember, its just the scale has big gaps between mob sizes. So a gargantuan sized object is actually smaller than a crab, cat or corgi, or something. Which is a bit incoherent when you have to reconcile those together visually?

And yeah, an atom flag would make sense for that. I'll make a PR for that then I guess.

Reply:
I don't care about mob vs. obj sizes in general, I just want to make sure this piece of UI code is doing what you want it to be doing.

Reply:
I was replying to loaf.

Reply:
But yeah, the name thing is a problem and its known. Its just there's no clean and reliable way to do that currently, short of just removing that adjective in front of the name afaik.

Reply:
Also, now that I look at this again, it probably won't be a problem for now. Since only human mobs are allowed to be wrapped currently.